lib/keila_web/live/create_user_live.ex

```diff
@@ -0,0 +1,36 @@
+defmodule KeilaWeb.CreateUserLive do
+  use KeilaWeb, :live_view
+  alias Keila.Auth.User
+
+  @impl true
+  def mount(_params, _session, socket) do
+    socket =
+      socket
+      |> assign(:changeset, Ecto.Changeset.change(%User{}))
+
+    {:ok, socket}
+  end
+
+  @impl true
+  def render(assigns) do
+    Phoenix.View.render(KeilaWeb.UserAdminView, "new.html", assigns)
+  end
+
+  @impl true
+  def handle_event("validate", params, socket) do
+    changeset =
+      %User{}
+      |> User.creation_changeset(params["user"])
+      |> then(fn changeset ->
+        if changeset.valid?,
+          do: changeset,
+          else: Ecto.Changeset.apply_action(changeset, :update) |> elem(1)
```

Comment:
Can we change the action to `:validate` just so that it's clearer this is to trigger error helper rendering?

Reply:
@glhrmv `:validate` is not an action that’s used by Ecto. https://hexdocs.pm/ecto/Ecto.Changeset.html#t:action/0

Reply:
It’s actually not necessary to use `apply_action/2` here at all. The docs suggest simply setting the `action` field directly: https://hexdocs.pm/ecto/Ecto.Changeset.html#module-changeset-actions

So this whole assignment could be simplified like this:

```elixir
changeset = 
  params["user"]
  |> User.creation_changeset()
  |> Map.replace!(:action, :insert)
```

Reply:
Hey @wmnnd thank for the response - I'm a greenhorn Phoenix developer so I really appreciate the links to the Ecto docs.

However, are you sure that code snippet is equivalent to what the MR is trying to do? It seems like the `:action` is being set only when the changeset is *invalid* - I thought the reason for this was to provide helpful error validation messages alongside the inputs in the UI, since Phoenix form error helpers only show errors if `:action` has a value, as per this section in the docs: https://hexdocs.pm/phoenix_live_view/1.0.0-rc.6/Phoenix.Component.html#form/1-a-note-on-errors

Reply:
The original code applies the action using `apply_action` which returns `{:ok, data}` if valid or `{:error, changeset}` if invalid. Note that the original code then returns the original changeset if `{:ok, data}` was returned and the updated changeset (with the action set) if `{:error, changeset}` was returned. So in both cases, we end up with a changeset. If the changeset was valid in the original code, we end up with a changeset without `action`, but for a valid changeset it makes no difference anyways.

Reply:
Thank you both, I'll change as @wmnnd mentioned!